assets/content/cookbook/Macros/strictly-typed-json.md

```diff
@@ -0,0 +1,41 @@
+[tags]: / "macro,json,expression-macro,configuration"
+
+# Strictly-Typed JSON
+
+It's possible read JSON files at compile-time into strictly-typed objects in haxe
+
+<img src="assets/haxe-json-macro.png" />
+
+Normally you might load a JSON file with something like this:
+```haxe
+var json = haxe.Json.parse(sys.io.File.getContent(path));
+```
+
+Instead, if you load the JSON in a macro, then the JSON data will be available at compile-time and therefore the types will be known:
+
+Create a file called **JsonMacro.hx** (or whatever you like) and add this:
+```haxe
+macro function load(path: String) {
```

Comment:
Might be worth pointing out that this can only work with constant paths.

Reply:
Added: https://github.com/HaxeFoundation/code-cookbook/pull/161/files#diff-5caa4672b0c39490a86919039f434c05ce51621a5026a642881040147e880467R39